src/credential-status/status-list.ts

```diff
@@ -0,0 +1,45 @@
+import * as zlib from 'pako'
+import { cborDecode, cborEncode } from '../cbor'
+import { type AllowedBitsPerEntry, StatusArray, allowedBitsPerEntry } from './status-array'
+
+export interface CborStatusListOptions {
+  statusArray: StatusArray
+  aggregationUri?: string
+}
+
+export class StatusList {
+  static buildCborStatusList(options: CborStatusListOptions): Uint8Array {
+    const compressed = options.statusArray.compress()
+
+    const statusList: Record<string, number | Uint8Array | string> = {
+      bits: options.statusArray.getBitsPerEntry(),
+      lst: compressed,
+    }
+
+    if (options.aggregationUri) {
+      statusList.aggregation_uri = options.aggregationUri
+    }
+    return cborEncode(statusList)
+  }
+
+  static verifyStatus(cborStatusList: Uint8Array, index: number, expectedStatus: number): boolean {
+    const statusList = cborDecode(cborStatusList) as Map<string, Uint8Array | number | string>
+    const bits = statusList.get('bits') as AllowedBitsPerEntry
+    const lst = statusList.get('lst') as Uint8Array
+
+    if (!statusList || !lst || !bits) {
+      throw new Error('Invalid status list format.')
```

Comment:
it would be good to define some errors, like we do for CoseError, but for the status list: https://github.com/animo/mdoc/blob/153111e46de98ebbebb585394ea27c282022fe2a/src/cose/error.ts#L11